operator/v1/types_ingress.go

```diff
@@ -125,6 +127,12 @@ type IngressControllerSpec struct {
 	//
 	// +optional
 	NodePlacement *NodePlacement `json:"nodePlacement,omitempty"`
+
+	// tlsSecurityProfile specifies settings for TLS connections for externally exposed servers.
+	//
+	// If unset, a default (which may change between releases) is chosen.
```

Comment:
This statement is ambiguous.  The default value for a new `IngressController` may change between releases, but does that change the effective value for an `IngressController` that was created with the default value?

Reply:
See https://github.com/openshift/api/pull/454#discussion_r330267806

Reply:
I now reference the ingress cluster config as the source of defaulting.

Reply:
That does not change anything.  If I create an ingresscontroller with the default TLS profile and then update the TLS profile in the ingress cluster config, does the ingresscontroller continue to use the old default, or should the ingress operator update the ingresscontroller's effective TLS profile to the new default value?

Reply:
Good question. I think that question applies to more than just this field.

I guess the semantics are up to us to define and document here on a case by case basis, because there is no coherent or enforceable rule applied broadly which would just answer the question for us.

How about:

>If unset, the default is based on the ingresses.config.openshift.io/cluster resource.
>Note that if the default is used, and the ingresses.config.openshift.io/cluster resource
>diverges from the ingresscontroller's effective default, the ingresscontroller will continue
>to use the defaults applied at the time of creation.

Reply:
Yeah, I like that because (1) it is consistent with other fields in `IngressControllerSpec`, (2) it avoids possibly confusing action at a distance, and (3) it is easiest to implement.

Reply:
Fixed